R/pecs.R

```diff
@@ -0,0 +1,139 @@
+#' @title Prediction Error Curves for PredictionSurv and LearnerSurv
+#' @description Methods to plot prediction error curves (pecs) for either a [PredictionSurv] object
+#' or a list of trained [LearnerSurv]s.
+#' @param x ([PredictionSurv] or `list` of [LearnerSurv]s)
+#' @param meas (`character(1)`) \cr
+#'   Either `"graf"` for [MeasureSurvGraf], or `"logloss"` for [MeasureSurvIntLogloss]
+#' @param times (`numeric()`) \cr
+#'   If provided then either a vector of time-points to evaluate `meas` or a range of time-points.
+#' @param n (`integer()`) \cr
+#'   If `times` is missing or given as a range, then `n` provide number of time-points to evaluate
+#'    `meas` over.
+#' @param eps (`numeric()`) \cr
+#'   Small error value to pass to [MeasureSurvIntLogloss] to prevent errors resulting from a log(0)
+#'   calculation.
+#' @param task ([TaskSurv])
+#' @param ... Additional arguments.
+#'
+#' @details If `times` and `n` are missing then `meas` is evaluated over all observed time-points
+#' from the [PredictionSurv] or [TaskSurv] object. If a range is provided for `times` without `n`,
+#' then all time-points between the range are returned.
+#'
+#' @examples
+#' library(mlr3)
+#' task = tsk("rats")
+#'
+#' # Prediction Error Curves for prediction object
+#' learn = lrn("surv.coxph")
+#' p = learn$train(task)$predict(task)
+#' pecs(p)
+#' pecs(p, meas = "logloss", times = c(20, 40, 60, 80)) +
+#'   ggplot2::geom_point() +
+#'   ggplot2::ggtitle("Logloss Prediction Error Curve for Cox PH")
+#'
+#' # Prediction Error Curves for fitted learners
+#' learns = lrns(c("surv.kaplan", "surv.coxph", "surv.ranger"))
+#' lapply(learns, function(x) x$train(task))
+#' pecs(learns, task = task, meas = "logloss", times = c(20, 90), n = 10)
+#'
+#'
+#' @export
+pecs = function(x, meas = c("graf","logloss"), times, n, eps = 1e-15, ...){
```

Comment:
is it possible to do this more general, i.e., user provides a custom measure. Maybe write out `measure`, or is `meas` a known term in the mlr3 world?

Reply:
how would that work when I want to plot pec for out-of-sample data?

Reply:
> is it possible to do this more general, i.e., user provides a custom measure.

the problem is that `Measure` objects have to return a single score, currently time-dependent measures don't work. so these are actually calling functions that i've written internally. To be honest as this function lives within `mlr3` I think it's okay to force users to use `mlr3` compatible measures, and until a time when I can figure out how to allow time-dependent measures to be allowed, then I think we are stuck with current design

> Maybe write out measure

will do

> how would that work when I want to plot pec for out-of-sample data? 

see below